google-account-plugin/src/com/google/cloud/tools/intellij/AccountPluginInitializationComponent.java

```diff
@@ -53,9 +56,26 @@ public void initComponent() {
     }
     if (!ApplicationManager.getApplication().isUnitTestMode()) {
       configureUsageTracking();
+      initUsageTracker();
     }
   }
 
+  @VisibleForTesting
+  void initUsageTracker() {
+    String platformPrefix = System.getProperty("idea.platform.prefix");
+    String trackingId = new PropertiesFilePluginFlags().getAnalyticsId();
+
+    Predicate<Object> userOptedIn = new Predicate<Object>() {
+      @Override
+      public boolean apply(@javax.annotation.Nullable Object object) {
+        return UsageTrackerManager.getInstance().hasUserOptedIn();
+      }
+    };
+
+    IdeaUsageTracker usageTracker = ServiceManager.getService(IdeaUsageTracker.class);
+    usageTracker.init(platformPrefix, trackingId, userOptedIn);
```

Comment:
Agreed. Unless this is Java 8, this is confusing. I will make this pass `PropertiesComponents`. A small downside is that it couples more tightly with IntelliJ, but this class is anyway bound to IntelliJ.